libraries/botbuilder/src/teamsActivityHandler.ts

```diff
@@ -924,4 +928,108 @@ export class TeamsActivityHandler extends ActivityHandler {
+                switch (context.activity.name) {
+                    case 'application/vnd.microsoft.meetingStart':
+                        return this.onTeamsMeetingStart(context);
+                    case 'application/vnd.microsoft.meetingEnd':
+                        return this.onTeamsMeetingEnd(context);
+                }
+            }
+
+            return super.dispatchEventActivity(context);
+        });
+    }
+
+    /**
+     * Invoked when a Meeting Started event activity is received from the connector.
+     * Override this in a derived class to provide logic for when a meeting is started.
+     *
+     * @param context The context for this turn.
+     * @returns A promise that represents the work queued.
+     */
+    protected async onTeamsMeetingStart(context: TurnContext): Promise<void> {
+        await this.handle(context, 'TeamsMeetingStart', this.defaultNextEvent(context));
+    }
+
+    /**
+     * Invoked when a Meeting End event activity is received from the connector.
+     * Override this in a derived class to provide logic for when a meeting is ended.
+     *
+     * @param context The context for this turn.
+     * @returns A promise that represents the work queued.
+     */
+    protected async onTeamsMeetingEnd(context: TurnContext): Promise<void> {
+        await this.handle(context, 'TeamsMeetingEnd', this.defaultNextEvent(context));
+    }
+
+    /**
+     * Registers a handler for when a Teams meeting starts.
+     *
+     * @param handler A callback that handles Meeting Start events.
+     * @returns A promise that represents the work queued.
+     */
+    public onTeamsMeetingStartEvent(
+        handler: (meeting: MeetingStartEventDetails, context: TurnContext, next: () => Promise<void>) => Promise<void>
+    ): this {
+        return this.on('TeamsMeetingStart', async (context, next) => {
+            const meeting = this.convertMeetingEventDetailsToCamelCase<MeetingStartEventDetails>(
+                context.activity.value
+            );
+            await handler(meeting, context, next);
+        });
+    }
+
+    /**
+     * Registers a handler for when a Teams meeting ends.
+     *
+     * @param handler A callback that handles Meeting End events.
+     * @returns A promise that represents the work queued.
+     */
+    public onTeamsMeetingEndEvent(
+        handler: (meeting: MeetingEndEventDetails, context: TurnContext, next: () => Promise<void>) => Promise<void>
+    ): this {
+        return this.on('TeamsMeetingEnd', async (context, next) => {
+            const meeting = this.convertMeetingEventDetailsToCamelCase<MeetingEndEventDetails>(context.activity.value);
+            await handler(meeting, context, next);
+        });
+    }
+
+    // The payload from Teams comes in with TitleCase keys, so we'll convert them to camelCase
+    // to maintain JSON and JS standard practices.
+    private convertMeetingEventDetailsToCamelCase<T extends MeetingStartEventDetails | MeetingEndEventDetails>(
```

Comment:
The other option I might consider is using `runtypes` to manage these scenarios.

```typescript
import * as t from "runtypes";

const TeamsMeetingStartT = t.Record({
  Id: t.String,
  ...,
  StartTime: t.String,
});

const TeamsMeetingEndT = t.Record({
  Id: t.String,
  ...,
  EndTime: t.String,
});

class TeamsActivityHandler {
  ...
  
  onTeamsMeetingEndEvent(handler) {
    return this.on('TeamsMeetingEnd', async (context, next) => {
      const meeting = TeamsMeetingEndT.check(context.activity.value);
      await handler({
        id: meeting.Id,
        ...,
        endTime: new Date(meeting.EndTime),
      }, context, next);
    });
  }
}
```

For me, the declarative style of `runtypes` is a little simpler and has the added benefit of useful and descriptive exceptions thrown if the data is truly malformed.